instrumentation/jaxrs/jaxrs-1.0/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxRsAnnotationsTracer.java

```diff
@@ -26,23 +27,24 @@
 
   public Span startSpan(Class<?> target, Method method) {
     String pathBasedSpanName = getPathSpanName(target, method);
-    Span serverSpan = BaseTracer.getCurrentServerSpan();
+    Context context = Context.current();
+    Span serverSpan = BaseTracer.getCurrentServerSpan(context);
 
     // When jax-rs is the root, we want to name using the path, otherwise use the class/method.
     String spanName;
     if (serverSpan == null) {
       spanName = pathBasedSpanName;
     } else {
       spanName = spanNameForMethod(target, method);
-      updateServerSpanName(serverSpan, pathBasedSpanName);
+      updateServerSpanName(context, serverSpan, pathBasedSpanName);
```

Comment:
I suspect we don't need this since we calculate the server span name correctly in the tracer IIUC. The fact that no tests changed means either this update isn't necessary or we're missing a test :)

Reply:
we didn't have any JAX-RS test that used a non-root context path, I added one now

Reply:
But you did not for JAX-RS 1.0 :)

Reply:
I added comment below about missing tests for jaxrs-1.0 and webflux